docs/core/v3.0.rst

```diff
@@ -0,0 +1,1761 @@
+The stored transformed data is restored to its original state whenever data is requested
+by the Array. Storage transformers can be configured per array via the
+`storage_transformers`_ name in the `array metadata`_. Storage transformers which do
+not change the storage layout (e.g. for caching) may be specified at runtime without
+adding them to the array metadata.
+
+A storage transformer serves the same `abstract store interface`_ as the store_.
+However, it should not persistently store any information necessary to restore the original data,
+but instead propagates this to the next storage transformer or the final store.
+From the perspective of an array or a previous stage transformer both store and storage transformer follow the same
+protocol and can be interchanged regarding the protocol. The behaviour can still be different,
+e.g. requests may be cached or the form of the underlying data can change.
+
+Storage transformers may be stacked to combine different functionalities:
+
+.. mermaid::
+
+    graph LR
+      Array --> t1
+      subgraph stack [Storage transformers]
+        t1[Transformer 1] --> t2[...] --> t3[Transformer N]
+      end
+      t3 --> Store
+
+A fixed set of storage providers is recommended for implementation with this specification:
+
+
+Predefined storage transformers
+-------------------------------
+
+- :ref:`sharding-storage-transformer-v1`
+
+Extensions
+==========
+
+
+Many types of extensions can exist and they can be grouped as
+following:
+
+======================= ========================================= =====================
+extension               metadata                                  is extension required
+======================= ========================================= =====================
+generic                 ``extensions`` in `entry point metadata`_ ``must_understand``
+array                   ``extensions`` in `Array metadata`_       ``must_understand``
+data type               `data_type`_                              no ``fallback``
+chunk grid              `chunk_grid`_                             always
+chunk memory layout     `chunk_memory_layout`_                    always
+storage transformer     `storage_transformers`_                   always
+======================= ========================================= =====================
+
+There are no group extensions in Zarr v3.0.
+
+See https://github.com/zarr-developers/zarr-specs/issues/49 for a list of potential extensions
+
+Implementation Notes
+====================
+
+This section is non-normative and presents notes from implementers about cases
+that need to be carefully considered but do not strictly fall into the spec.
+
+Explicit vs implicit group
+--------------------------
+
+While the zarr spec v3 defined implicit and explicit group, implementation may
+decide to create an explicit group for all implicit group they encounter; in
+particular when using a hierarchical storage.
```

Comment:
Ok, I think this addresses the comment I raised below. Maybe we should mention that implicit groups do not ensure that there won't be a race condition?

Reply:
See my comment above, do you think 2040c59 resolves this?

Reply:
Yep, sounds good.